### ISSUE TITLE
Place Picker doesn't work

### ISSUE THREAD

**Issue by drmrbrewer** [microg/android_packages_apps_GmsCore#287]
My app is compiled with the latest versions of the Play services libraries, and makes use of the Place Picker https://developers.google.com/places/android-api/placepicker

I have a user who reports that the Place Picker doesn't work as expected with micrig.  It's difficult to provide much detail, as I don't have microg myself, but is this a known issue?  If I revert to an earlier version of Play services in my app (9.6.1 I think) then Place Picker apparently works fine.

**Comment by ale5000-git** [comment]
@drmrbrewer: I think it would help if you provide your app or another app that have the same problem.

**Comment by drmrbrewer** [comment]
@ale5000-git it's [this one](https://play.google.com/store/apps/details?id=com.cloud3squared.meteogram).  Place a widget on the home screen, and in the widget settings page, click on the Location heading to open up the location settings, then click on the Text button for a Place Autocomplete or on the Map button for a Place Picker.  Apparently neither work.  It seems that the Place Picker map dialog appears, but the user cannot actually get any coordinates other than the default location to be returned.  See attached screenshot from them.

![screenshot_20170128-172109](https://cloud.githubusercontent.com/assets/9009792/22404574/51658c0e-e62b-11e6-9df4-d604e7a6a46d.png)

**Comment by mar-v-in** [comment]
@drmrbrewer can't you move the map to choose a different location and then touch the "select this location" button (it doesn't look like a button because of the theming...)?

**Comment by drmrbrewer** [comment]
@mar-v-in  the Place Picker interface works fine on my own test devices, because I'm not running microg.  So I'm relating a user's problem second-hand here.  From their screenshot, it seems from the "Can't load nearby places" that the Place Picker (provided entirely by Google of course) isn't working as it should.  And they can't seem to get any coordinates to be returned (to the location page of my app) other than for the default location.

Does it work for you?  Presumably you're running microg, right?

**Comment by mar-v-in** [comment]
You can not select POI from nearby places in microG so far, only by directly pointing on them using the map.

**Comment by drmrbrewer** [comment]
So "Select this location" (see screenshot) should work?

**Comment by mar-v-in** [comment]
Yes.

**Comment by drmrbrewer** [comment]
OK I'll inform the user and check whether it does work or not.

What about the text-based autocomplete lookup (from the Text button in my app)... does that work OK for you?

**Comment by mar-v-in** [comment]
No, text based autocomplete won't work, Places functionality is missing, other than the location picking (= not place picking)

**Comment by drmrbrewer** [comment]
Ah OK.  Any plans in place to make autocomplete work?  I'm pretty sure this user was reporting that it was working fine when I was using play services 9.6.1.  And what do you mean by location picking vs place picking?

**Comment by mar-v-in** [comment]
There are no difference with this in older versions I am aware of.

Location picking means that the returned "place" object is not populated with informative fields (proper name or address), which would be the case when using original play services. For a weather app this should not make a huge difference if you use the location values (latitude+longitude).

**Comment by drmrbrewer** [comment]
Ah I see.  Yes latitude and longitude is all that is really needed to get basic functionality in my app (better to have place name etc but not essential).  

So with the map-based location picker, what information is returned into the place name etc fields of my location section?  If I can detect that valid info isn't present on return from the place picker, I should be able to do a geolocation lookup based on lat long.  Maybe.

**Comment by ale5000-git** [comment]
@mar-v-in: Maybe I have misunderstood something but isn't the task of Nominatim Geocoder Backend to return the "proper name or address"?

**Comment by xnisyaj** [comment]
Hi, i'm having the same issue. I cant pick places in any app since i installed MicroG. Any app that require Place Picker just don't let me long-press for selecting a location. Long-press just don't do anything. I can find any error on Logcat.

**Comment by mar-v-in** [comment]
@xnisyaj you need to move the map under the pinpoint so that the pinpoint points to the location you want to share. Then touch "select this location" to use it.

**Comment by xnisyaj** [comment]
@mar-v-in the pinpoint in that apps is not shown until you long-press. And since i can't long press, the pinpoint is never shown. I tried it on a lot of apps (fake gps apps, gps alarm apps, etc) and it's not working in any of it. The only working app is "com.lexa.fakegps" that have, like you said, a fixed pinpoint on the center. And i can press "select this location".

**Comment by mar-v-in** [comment]
ah, ok you're talking about a completely different problem: This is about the place picker, a component provided by microG directly.

The apps you mentioned use the maps api to build their own place picker. Can you please open a new issue about maps api not supporting long press listeners, stating at least one app that does not work (with full package name). Thanks.

**Comment by OhSoGood** [comment]
Hi, I hope I incorrectly hijack the thread. A free app I'd like to use, ChargeMap, throw an exception with this logcat message: No Activity found to handle Intent { act=com.google.android.gms.location.places.ui.AUTOCOMPLETE pkg=com.google.android.gms
Is that the same issue as stated above? Any change to solve it?
